temporian/implementation/numpy/operators/assign.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation for the Assign operator."""
+from typing import Dict, List
+
+import numpy as np
+from temporian.implementation.numpy.data.event import NumpyEvent
+from temporian.implementation.numpy.data.event import NumpyFeature
+from temporian.implementation.numpy.data.sampling import NumpySampling
+
+
+def _get_common_timestamps(
+    sampling_1: NumpySampling, sampling_2: NumpySampling, index: tuple
+) -> np.ndarray:
+    """It returns a np.ndarray with same shape as sampling_1.data[index] with True
+    values where sampling_1 and sampling_2 have the same timestamp.
+
+    Args:
+        sampling_1 (NumpySampling): sampling of assignee event.
+        sampling_2 (NumpySampling): sampling of assigned event.
+        index (tuple): index to check.
+
+    Returns:
+        np.ndarray: boolean np.darray with same shape as sampling_1.data[index]
+
+    Examples:
+        >>> sampling_1 = NumpySampling(
+        ...     names=["user_id"],
+        ...     data={
+        ...         (151591562,): np.array(
+        ...             ["2022-02-05", "2022-02-06", "2022-02-07"],
+        ...             dtype="datetime64",
+        ...         ),
+        ...         }
+        ...     )
+        >>> sampling_2 = NumpySampling(
+        ...     names=["user_id"],
+        ...     data={
+        ...         (151591562,): np.array(
+        ...             ["2022-02-06", "2022-02-07", "2022-02-08"],
+        ...             dtype="datetime64",
+        ...         ),
+        ...         }
+        ...     )
+        >>> _get_common_timestamps(sampling_1, sampling_2, (151591562,))
+        array([False, True, True])
+
+    """
+    return np.isin(sampling_1.data[index], sampling_2.data[index])
+
+
+def _convert_feature_to_new_sampling(
+    feature: NumpyFeature,
+    common_timestamps: List[bool],
+    new_sampling: NumpySampling,
+    index: tuple,
+):
+    """Convert feature to new sampling. It requires a boolean list with same length
+    as the new sampling, where True values indicate that the feature has a value
+    for that timestamp. And it needs the new sampling to assign to the new Feature object.
+
+    Args:
+        feature (NumpyFeature): feature to convert.
+        common_timestamps (List[bool]): list of booleans indicating if sampling_2 has same timestamp in sampling_1
```

Comment:
We haven't named `sampling_2` and `sampling_1` in this docstring yet - I'm assuming they are `new_sampling` and the feature's sampling respectively, correct? In that case, we should reflect that in the documentation.